src/mem/mem_ctrl.hh

```diff
@@ -106,8 +105,12 @@ class MemPacket
 
     const bool read;
 
+    //every write will initiate a read first
```

Comment:
Needs more comment here, at least if this is going to be the permanent way this is implemented.

Also, should probably be const